### ISSUE TITLE
Laker.bayes Latin-hypercube sampling uses unseeded np.random, breaking seed-based reproducibility

### ISSUE THREAD

**Issue by sachncs** [sachncs/laker#17]
## Problem

`Search.bayes` uses `np.random.permutation` and `np.random.uniform` (the global numpy RNG) to draw Latin-hypercube candidates, so the `seed` argument does not make the BO trajectory reproducible. This is the documented `seed` parameter on the public `Laker.bayes` method.

## Evidence

`laker/search.py:182-188`:

```python
def lh_sample(s: int) -> np.ndarray:
    d = bounds.shape[0]
    samples = np.zeros((s, d), dtype=np.float64)
    for i in range(d):
        perm_np = np.random.permutation(s)         # global RNG
        samples[:, i] = (perm_np + np.random.uniform(size=s)) / s   # global RNG
    return samples * (bounds[:, 1] - bounds[:, 0]) + bounds[:, 0]
```

The `seed` argument is forwarded to `Laker.fit` (line 209) but is never used to seed numpy. `torch.Generator` is used to seed the train/val split at line 167-168, so that part *is* reproducible; only the LH candidate draws are not.

By contrast, `Math.latin` (`laker/math.py:73-79`) does the right thing — it takes a `seed` and constructs `rng = np.random.default_rng(seed)`. The bayes code bypasses that helper.

## Root Cause

The bayes candidate generator was rolled inline in `search.py` instead of calling `Math.latin`. The author seeded the train/val split (which is what gets passed to `fit` at the end) but missed seeding the candidate pool.

## Impact

Calling `Laker.bayes(x, y, seed=42)` twice produces two different acquisition histories and, depending on the GP landscape, two different best parameters. Users who depend on `seed` for reproducibility (e.g. hyperparameter sweeps, regression tests) get a different result each run.

This is a contract violation against the documented `seed` parameter on `Laker.bayes` (`laker/model.py:481-504`).

## Recommendation

Either:

1. Use the existing `Math.latin` helper inside `Search.bayes` (parameterise the per-dimension sample count), or
2. Add a `numpy_random_generator` argument to `lh_sample` and pass one seeded from the public `seed`.

Add a regression test that calls `bayes` twice with the same `seed` and asserts the returned `best_params` are equal.

## Acceptance Criteria

- [ ] `Laker.bayes(x, y, seed=42)` called twice in the same process returns identical `best_params`.
- [ ] A test in `tests/test_search.py` covers this.
- [ ] `Search.bayes` no longer calls `np.random.permutation` / `np.random.uniform` directly.

## Verification

```python
import numpy as np
m1 = Laker(verbose=False)
m2 = Laker(verbose=False)
m1.bayes(x, y, n_calls=5, n_init=3, seed=42)
m2.bayes(x, y, n_calls=5, n_init=3, seed=42)
assert m1.lam == m2.lam and m1.gamma == m2.gamma
```

## Dependencies

None.

## Risk

Low. The fix is internal; the public surface is unchanged.

**Comment by sachncs** [comment]
Resolved in PR #32.
